plugins/discovery/pod.go

```diff
@@ -27,28 +27,39 @@ func newPodHandler(kubeClient kubernetes.Interface, discoverer discovery.Discove
 
 	inf.AddEventHandler(cache.ResourceEventHandlerFuncs{
 		AddFunc: func(obj interface{}) {
-			pod := obj.(*v1.Pod)
-			podUpdated(pod, discoverer)
+			updatePodIfValid(obj, discoverer)
 		},
 		UpdateFunc: func(_, obj interface{}) {
-			pod := obj.(*v1.Pod)
-			podUpdated(pod, discoverer)
+			updatePodIfValid(obj, discoverer)
 		},
 		DeleteFunc: func(obj interface{}) {
-			pod := obj.(*v1.Pod)
-			discoverer.Delete(discovery.Resource{
-				Kind:       discovery.PodType.String(),
-				IP:         pod.Status.PodIP,
-				Meta:       pod.ObjectMeta,
-				Containers: pod.Spec.Containers,
-			})
+			deletePodIfValid(obj, discoverer)
 		},
 	})
 	return &podHandler{
 		informer: inf,
 	}
 }
 
+func deletePodIfValid(obj interface{}, discoverer discovery.Discoverer) {
+	if _, ok := obj.(cache.DeletedFinalStateUnknown); !ok {
```

Comment:
Should we be deleting the pod in this case anyway?

Reply:
I updated to type assert on Pod. I don't think we want to try to delete, because we won't have any pod information to use in this case. In the library code, they return a nil object if they hit this